webapp/app/js/models/repository.js

```diff
@@ -95,8 +95,35 @@ treeherder.factory('ThRepositoryModel', [
 
             // return the promise of getting the repos
             return get_list().
-                success(function (data) {
-                    $rootScope.repos = data;
+                success(function(data) {
+                    // FIXME: only supporting github + hg for now for pushlog
+                    // + revision info
+                    function Repo(props) {
+                        _.assign(this, props);
+                        if (this.url.indexOf('github') >= 0) {
```

Comment:
I have a slight preference for using the dvcs_type field and hinging this all off of 'hg' vs 'git' rather than searching for substrings, but up to you.